src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -321,7 +321,7 @@ def load_arguments(self, _):
                        help="The eviction policy for the Spot priority virtual machine. Default eviction policy is Deallocate for a Spot priority virtual machine")
         c.argument('enable_agent', arg_type=get_three_state_flag(), min_api='2018-06-01',
                    help='Indicates whether virtual machine agent should be provisioned on the virtual machine. When this property is not specified, default behavior is to set it to true. This will ensure that VM Agent is installed on the VM so that extensions can be added to the VM later')
-
+        c.argument('guest_provision_signal', arg_type=get_three_state_flag(), help='Specifies whether the guest provision signal is required to infer provision success of the virtual machine.')
```

Comment:
I'd prefer `require_guest_provision_signal`. Without "require", the mean is vague and not intuitive.